distribution/sdk/sample-mpegts-stream-generator/src/main/java/org/codice/alliance/distribution/sdk/video/stream/mpegts/MpegTsUdpClient.java

```diff
@@ -0,0 +1,139 @@
+import org.slf4j.LoggerFactory;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.DatagramPacket;
+import io.netty.channel.socket.nio.NioDatagramChannel;
+
+/**
+ * This client is used for testing/development to transmit an MPEG-TS file as a stream of UDP
+ * packets.
+ */
+public class MpegTsUdpClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MpegTsUdpClient.class);
+
+    public static void main(String[] args) {
+
+        if(args.length != 1) {
+            LOGGER.error("Cannot start Mpegts stream.  No file specified.");
+        }
+
+        String videoFilePath = args[0];
+
+        LOGGER.info("Video file path : {}", videoFilePath);
+
+        final AtomicLong count = new AtomicLong(0);
+
+        EventLoopGroup eventLoopGroup = new NioEventLoopGroup();
+        try {
+
+            Bootstrap bootstrap = new Bootstrap();
+
+            bootstrap.group(eventLoopGroup)
+                    .channel(NioDatagramChannel.class)
+                    .option(ChannelOption.SO_BROADCAST, true)
+                    .handler(new SimpleChannelInboundHandler<DatagramPacket>() {
+                        @Override
+                        protected void channelRead0(ChannelHandlerContext channelHandlerContext,
+                                DatagramPacket datagramPacket) throws Exception {
+                            LOGGER.info("channelRead0");
+                        }
+
+                        @Override
+                        public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause)
+                                throws Exception {
+                            LOGGER.error("error", cause);
+                            ctx.close();
+                        }
+                    });
+
+            Channel ch = bootstrap.bind(0)
+                    .sync()
+                    .channel();
+
+            long bytesSent = 0;
+
+            long tsPacketCount = 618570;
+            long tsDurationMillis = (4 * 60 + 39) * 1000;
+
+            double delayPerPacket = (double) tsDurationMillis / (double) tsPacketCount;
+
+            long startTime = System.currentTimeMillis();
+
+            int packetsSent = 0;
+
+            try (final FileInputStream fis = new FileInputStream(videoFilePath)) {
+
+                byte[] buffer = new byte[188];
+                int c;
+                while ((c = fis.read(buffer)) != -1) {
+                    bytesSent += c;
+
+                    ChannelFuture cf = ch.writeAndFlush(new DatagramPacket(Unpooled.copiedBuffer(
+                            buffer), new InetSocketAddress("127.0.0.1", 50000)));
```

Comment:
Should the IP address and port number be made into command line args with sensible defaults?